File: 2025_3/pr_12924_ver2.py
def solution(n):
    left = 0
    cnt = 0
    total = 0
    for i in range(1, n+1):
        total += i
        while total > n and left < i:
            total -= left
            left += 1
            
        if total == n:
            cnt += 1  
    return cnt

File: 2025_3/pr_12924_ver2.js
function solution(n) {
    let ans = 0
    for(let i=1; i<(n+1); i++) {
        let total = 0;
        for(let j=i; j< n + 1; j++) {
            total += j
            
            if (total == n) {
                ans += 1
                break
            } else if (total > n) {
                break
            }
                
        }
    }
    return ans
}